cou/steps/hypervisor.py

```diff
@@ -29,23 +31,49 @@
 logger = logging.getLogger(__name__)
 
 
-@dataclass(frozen=True)
-class HypervisorMachine:
-    """Hypervisor machine containing units for multiple applications."""
-
-    name: str
-    units: list[COUUnit]
-
-
-@dataclass(frozen=True)
+@dataclass
 class HypervisorGroup:
     """Group of hypervisors.
 
     For example, this can represent a group of all hypervisors in a single availability zone.
     """
 
     name: str
-    hypervisors: list[HypervisorMachine]
+    app_units: dict[str, list[COUUnit]]
+
+    def __eq__(self, other: Any) -> bool:
+        """Equal magic method for HypervisorGroup.
+
+        :param other: HypervisorGroup object to compare.
+        :type other: Any
+        :return: True if equal False if different.
+        :rtype: bool
+        """
+        if not isinstance(other, HypervisorGroup):
+            return NotImplemented
+
+        return other.name == self.name
+
+
+class AZs(defaultdict):
```

Comment:
I think we should change this class name to `UpgradeGroups`. reviewing other PRs I realized that this name it's confusing

Reply:
We can rename it any time, resolving this comment to unblock my work